instaclustr/resource_cluster.go

```diff
@@ -323,45 +364,56 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 	d.SetId(id)
 	d.Set("cluster_id", id)
 	log.Printf("[INFO] Cluster %s has been created.", id)
-	return nil
+
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```

Comment:
Could the retry logic be optional and timeout length be configurable as arguments? 

I've seen clusters taking more time than 15 mins (on GCP for example) quite often, and by having the retry logic as optional there's some flexibility in how the plan can be applied. Some build pipelines may want to just fire and continue.

Since this fix is meant for the `kafka_user` resource, could that resource be made to wait for the cluster to reach the RUNNING status? See `resourceVpcPeeringCreate` for example.

Reply:
The timeout can be adjusted [timeout block](https://www.terraform.io/docs/configuration/resources.html#operation-timeouts) like:
```hcl
resource "instaclustr_cluster" "example" {
  ...
  timeouts {
    create = "60m"
  }
}
```

I think that I can create a `wait` property which can be used to trigger this behaviour.

This fix was not only for the `kafka_user`. In my case, I'm using Cassandra cluster. I need:
- `private_contact_addresses` to setup DNS entries,
- `data_centre_id` to create VPC peering in GCP
- `cluster_certificate_download`, `username` and `instaclustr_user_password` for another provider which setup users and keyspaces.

I don't know if all those values are available in `PROVISIONED` state. Isn't is safer to wait for `RUNNING` state?

Reply:
For the kafka user resource, we'd have to wait for `RUNNING` that's right. Though with the wait property it won't be necessary. We can resolve the dependents waiting in another PR. 

Thanks for pointing out the adjustable wait definition, that works great. A wait property would be perfect and give us a decent amount of flexibility, thanks!

Reply:
It might be useful to have the cluster condition as a parameter also. 
In that way if you are only waiting for the ip addresses, you can wait for the PROVISIONED state, however if you need to add kafka users, or other items, then you can wait for a running state?